Changed static forms to Dialog boxes

peterhsieh opened the pull request:
Removed call button from parent component and attached to form
component. Changes made to Payment and Repair Requests.

peterhsieh committed:
Changed static forms to Dialog boxes

Removed call button from parent component and attached to form
component. Changes made to Payment and Repair Requests.

nirrek commented on scripts/components/RepairRequestForm.jsx:
```diff
@@ -73,32 +77,35 @@ var RepairRequestForm = React.createClass({
+            name="Description"
+            onChange={this.onChange.bind(this, 'description')}
+            floatingLabelText="Description" />
+          <div style={style.inputContainer}>
```
Update this to use the `<ImageSelector>` component I made. I mentioned this to the team in the chat the other day. This component supports things like specifying a maximum file size allowed, and will allow for us to add features to the component if necessary. We need to have an image upload control in tonnes of places, so it will be easier to maintain if they all are using a single component.

peterhsieh replied:
Oh sorry! Totally forgot! Updated :)

nirrek commented on scripts/components/RepairRequestForm.jsx:
```diff
@@ -73,32 +77,35 @@ var RepairRequestForm = React.createClass({
     var errorMessage;
+    var standardActions = [
+      { text: 'Cancel' },
+      { text: 'Submit', onTouchTap: this.onSubmit, ref: 'submit' }
```
Try not to use `Submit` as the action for things, as it conveys very little meaning (especially so as it's the default name for any form ever created). Instead describe what action the button performs. This comment form I am currently writing in on Github is a good example. The action says `Comment`, it does not say `Submit`. Similarly, for lodging the repair request say `Lodge Repair Request`, or something like that. It's a nice usability win, and it's very simple to implement once you are in the habit of actually describing what your interface does.

peterhsieh replied:
Yep! I agree! Will keep that in mind :+1:

peterhsieh committed:
Form Usability, updated to Image Selector module

Form Usability
- Changed submit buttons to more relevant contexts

Image Selector Module
- Now used in RRequest form
- Changed attribute name of dataUri to image for better code readability

peterhsieh commented:
Both comments implemented.
Had to do a small change on server code. See PR - https://github.com/nirrek/wallflyServer/pull/7

nirrek commented:
@PeterHsieh The date selector positioning is screwed when the payments form is a modal (it's going off the bottom of my screen). Can you reproduce this behaviour?
![image](https://cloud.githubusercontent.com/assets/1008618/10006944/d3ae1ac6-6104-11e5-9475-d590a2f1d2cb.png)

Also, I don't think the current date picker is usable for this form. Try to fill the form out with details of one of your credit cards, and see what problem you run into.

nirrek commented on scripts/components/RepairRequestForm.jsx:
```diff
@@ -50,55 +55,62 @@ var RepairRequestForm = React.createClass({
+          <div style={style.error}> { errorMessage } </div>
+          <TextField
+            value={description}
+            multiLine={true}
```
This field is intended to have a paragraph written in it. Can you use the `fullWidth` property so that it fills the available space for writing.

Also, similar to not using `Submit` to describe a button, we also shouldn't use `Description` to describe the purpose of a field where we want complex user input. We want to guide the user with using the form field correctly. A possible improvement would be:

`floatingLabelText="Describe what needs repairing, how urgent it is, and how the damage occurred."`

peterhsieh commented:
Mm yep - I can reproduce this issue.
I see what you mean, CC expiry dates are well ahead current date. Maybe it should be two dropdown menus for month and year?

nirrek commented:
Yeh. Or we could just have them type it in as it will only be 4 digits. So
have inputs like: `[07] / [15]`, where the things in brackets are separate
inputs that only accept 2 characters.

On Tue, Sep 22, 2015 at 8:57 AM, PeterHsieh notifications@github.com
wrote:

> Mm yep - I can reproduce this issue.
> I see what you mean, CC expiry dates are well ahead current date. Maybe it
> should be two dropdown menus for month and year?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/nirrek/wallfly/pull/14#issuecomment-142131953.

peterhsieh committed:
RRequest Description Usability, Payment Form date picker removed, simple text inputs added

nirrek committed:
(PaymentForm) cleanup form styles a bit
